code/modules/antagonists/traitor/components/demoraliser.dm

```diff
@@ -0,0 +1,154 @@
+ * - Applies a mood to people who are in visible range of the item.
+ * - Does not re-apply mood to people who already have it.
+ * - Sends a signal if a mood is successfully applied.
+ */
+/datum/proximity_monitor/advanced/demoraliser
+	var/datum/demoralise_moods/moods
+
+/datum/proximity_monitor/advanced/demoraliser/New(atom/_host, range, _ignore_if_not_on_turf = TRUE, datum/demoralise_moods/moods)
+	. = ..()
+	src.moods = moods
+	RegisterSignal(host, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/proximity_monitor/advanced/demoraliser/Destroy()
+	UnregisterSignal(host, COMSIG_PARENT_EXAMINE)
+	return ..()
+
+/datum/proximity_monitor/advanced/demoraliser/field_turf_crossed(atom/movable/crossed, turf/location)
+	if (!isliving(crossed))
+		return
+	if (!can_see(crossed, host, current_range))
+		return
+	on_seen(crossed)
+
+/*
+ * Signal proc for [COMSIG_PARENT_EXAMINE].
+ * Immediately tries to apply a mood to the examiner, ignoring the proximity check.
+ * If someone wants to make themselves sad through a camera that's their choice I guess.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/on_examine(datum/source, mob/examiner)
+	SIGNAL_HANDLER
+	if (isliving(examiner))
+		on_seen(examiner)
+
+/**
+ * Called when someone is looking at a demoralising object.
+ * Applies a mood if they are conscious and don't already have it.
+ * Different moods are applied based on whether they are an antagonist, authority, or 'other' (presumed crew).
+ *
+ * Arguments
+ * * viewer - Whoever is looking at this.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/on_seen(mob/living/viewer)
+	if (!viewer.mind)
+		return
+	// If you're not conscious you're too busy or dead to look at propaganda
+	if (viewer.stat != CONSCIOUS)
+		return
+	if (was_demoralised(viewer))
+		return
+
+	if (is_special_character(viewer))
+		to_chat(viewer, span_notice("[moods.antag_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.antag_mood)
+	else if (viewer.mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND))
+		to_chat(viewer, span_notice("[moods.authority_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.authority_mood)
+	else
+		to_chat(viewer, span_notice("[moods.crew_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.crew_mood)
+
+	SEND_SIGNAL(host, COMSIG_DEMORALISING_EVENT, viewer.mind)
+
+/**
+ * Returns true if the viewer already has been given feelings, false if they haven't.
+ *
+ * Arguments
+ * * viewer - Whoever just saw the parent.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/was_demoralised(mob/living/viewer)
+	var/datum/component/mood/mood = viewer.GetComponent(/datum/component/mood)
+	if (!mood)
+		return FALSE
+
+	for(var/i in mood.mood_events)
+		var/datum/mood_event/moodlet = mood.mood_events[i]
+		if (moodlet.category == moods.mood_category)
+			return TRUE
+
+	return FALSE
```

Comment:
This would make sense on the mood component and I guess as I am already getting the component it wouldn't harm to move it there.
I have seen that someone else is apparently datumising the mood system which would also make this make more sense than how it is presently.